packages/http/src/validator/__tests__/index.spec.ts

```diff
@@ -51,150 +43,112 @@ describe('HttpValidator', () => {
+          );
         });
       });
     });
+  });
 
-    describe('query validation in enabled', () => {
-      const test = (extendResource?: Partial<IHttpOperation>, extendInput?: Partial<IHttpRequest>) => () => {
-        jest
-          .spyOn(resolveValidationConfigModule, 'resolveRequestValidationConfig')
-          .mockReturnValueOnce({ body: false, headers: false, hijack: false, query: true });
-
-        expect(
-          httpValidator.validateInput({
-            resource: Object.assign(
-              {
-                method: 'get',
-                path: '/',
-                id: '1',
-                request: {},
-                responses: [{ code: '200' }],
-              },
-              extendResource,
-            ),
-            input: Object.assign({ method: 'get', url: { path: '/', query: {} } }, extendInput),
-            config: { cors: false, mock: { dynamic: false }, validate: { request: { query: true } } },
-          }),
-        ).toEqual([mockError]);
+  describe('headers validation in enabled', () => {
+    const test = (extendResource?: Partial<IHttpOperation>, length: number = 1) => () => {
+      expect(
+        httpValidator.validateInput({
+          resource: Object.assign(
+            {
+              method: 'get',
+              path: '/',
+              id: '1',
+              request: {},
+              responses: [{ code: '200' }],
+            },
+            extendResource,
+          ),
+          input: { method: 'get', url: { path: '/' } },
+          config: { cors: false, mock: { dynamic: false }, validateRequest: true, validateResponse: true },
+        }),
+      ).toHaveLength(length);
+    };
+
+    describe('request is not set', () => {
+      it('validates headers', test(undefined, 2));
+    });
+  });
 
-        expect(resolveValidationConfigModule.resolveRequestValidationConfig).toHaveBeenCalled();
-        expect(httpBodyValidator.validate).not.toHaveBeenCalled();
-        expect(httpHeadersValidator.validate).not.toHaveBeenCalled();
-        expect(httpQueryValidator.validate).toHaveBeenCalledWith({}, [], undefined);
-      };
+  describe('query validation in enabled', () => {
+    const test = (
+      extendResource?: Partial<IHttpOperation>,
+      extendInput?: Partial<IHttpRequest>,
+      lenght: number = 2,
+    ) => () => {
+      expect(
+        httpValidator.validateInput({
+          resource: Object.assign(
+            {
+              method: 'get',
+              path: '/',
+              id: '1',
+              request: {},
+              responses: [{ code: '200' }],
+            },
+            extendResource,
+          ),
+          input: Object.assign({ method: 'get', url: { path: '/', query: {} } }, extendInput),
+          config: { cors: false, mock: { dynamic: false }, validateRequest: true, validateResponse: true },
+        }),
+      ).toHaveLength(lenght);
```

Comment:
Nothing to do with this PR @XVincentX but I think we should compare to what the actual tests are not a length. Could be a good one to refactor later.